sdk/helper/ldaputil/client.go

```diff
@@ -85,6 +86,10 @@ func (c *Client) DialLDAP(cfg *ConfigEntry) (Connection, error) {
 		retErr = multierror.Append(retErr, errwrap.Wrapf(fmt.Sprintf("error connecting to host %q: {{err}}", uut), err))
 	}
 
+	if timeout := cfg.RequestTimeout; timeout > 0 {
```

Comment:
Just curious, why use the `timeout` variable here instead of the request timeout directly? Not a big deal, just hadn't seen it before so wanted to hear your perspective.

Reply:
Mostly a preference of using a scoped variable if the value used in the conditional check is also used inside the conditional itself.

I think there are a few cases in tests where something similar happens:
```
if exp := 0; code != exp {
	t.Errorf("expected %d to be %d", code, exp)
}
```

I'm good with referencing `cfg.RequestTimeout` twice though and can update if that's preferred.

Reply:
Nope, that's fine. Don't need to change it. Truly just curious about situations where I might want to choose it myself.

Is there a benefit to using a scoped variable? Is it that `cfg.RequestTimeout` is only read out once, so it's less operations? Or is it more like you're potentially capturing its state at the time of the check and then being sure you're using the same state below?

Reply:
The benefit I see is ease of updating the value. In the example above, if I were to update the expected value `0`, I would have to do it in multiple places. 

If timeout happens to be something else in the future, I'd have to update the value in the conditional check as well as the usage inside the conditional. In most cases the scoping is trivial (as is with my actual use case) so it's unlikely that I forget to update. However I'd be willing to sacrifice a little bit of overhead to prevent potential bugs from happening.

Reply:
That makes sense! Thanks for the explanation.